### PR TITLE
Handle undefined callback with empty path params

### PR DESCRIPTION
**Pull request by kvigen**
Missed the empty callback case

### CODE REVIEW

File: clients/js/genjs.go [hunk 1]
```diff
@@ -246,7 +246,9 @@ var methodTmplStr = `
     const headers = {};
     {{- range $param := .PathParams}}
     if (!params.{{$param.JSName}}) {
-      cb(new Error("{{$param.JSName}} must be non-empty because it's a path parameter"))
+      if (cb) {
+        cb(new Error("{{$param.JSName}} must be non-empty because it's a path parameter"));
+      }
```
Comment (cozmo) [inline clients/js/genjs.go]: I think this isn't quite right - I think we want to call the callback if it exists, else we want to reject the promise. That's what the `resolver` and `rejector` functions are for below, but those are in the promise. Not sure exactly how we want to solve this but we may just want to wrap the whole function body with the promise and then we can use `resolve` and `reject` like we do below?

Comment (kvigen) [inline clients/js/genjs.go]: derp. yeah, wrapping the whole function body makes sense

Comment (kvigen) [inline clients/js/genjs.go]: @cozmo back to you. what do you think?

**Comment by cozmo** [comment]
Yea this looks good to me.

**Pull request merged**